RecoEgamma/EgammaPhotonProducers/src/GEDPhotonProducer.cc

```diff
@@ -503,17 +506,21 @@ void GEDPhotonProducer::fillPhotonCollection(edm::Event& evt,
 	ptFast(parentSCRef->energy(),parentSCRef->position(),math::XYZPoint(0,0,0)) <= preselCutValues[0] ) continue;
     // calculate HoE    
 
-    const CaloTowerCollection* hcalTowersColl = hcalTowersHandle.product();
+    const CaloTowerCollection* hcalTowersColl = nullptr;
+    if (not hcalTowers_.isUninitialized()){
+      hcalTowersColl = hcalTowersHandle.product();
+    }
     EgammaTowerIsolation towerIso1(hOverEConeSize_,0.,0.,1,hcalTowersColl) ;  
     EgammaTowerIsolation towerIso2(hOverEConeSize_,0.,0.,2,hcalTowersColl) ;  
-    double HoE1=towerIso1.getTowerESum(&(*scRef))/scRef->energy();
-    double HoE2=towerIso2.getTowerESum(&(*scRef))/scRef->energy(); 
+    double HoE1 = (hcalTowersColl != nullptr ? towerIso1.getTowerESum(&(*scRef))/scRef->energy() : 0.f);
+    double HoE2 = (hcalTowersColl != nullptr ? towerIso2.getTowerESum(&(*scRef))/scRef->energy() : 0.f); 
```

Comment:
for all of this : why not simply
```
double HoE1,HoE2;
HoE1=HoE2=0;
if (not hcalTowers_.isUninitialized()){
//all old implementation
}
```
this way there is also no need to modify EgammaTowerIsolation